wit/guest.wit

```diff
@@ -1,10 +1,7 @@
-interface messaging-guest {
-    use messaging-types.{message, guest-configuration, error};
+interface guest {
+    use types.{message, guest-configuration, error};
 
-    /// Returns the list of channels (and extension metadata within guest-configuration) that 
-    /// this component should subscribe to and be handled by the subsequent handler within guest-configuration
-    configure: func() -> result<guest-configuration, error>;
```

Comment:
Configuration should be provided by the host to the guest via runtime-config or be handled by the messaging implementation itself. This matches how other things are done right now as well in keyvalue